packages/material-ui/src/index.d.ts

```diff
@@ -64,6 +64,42 @@ export type ConsistentWith<T, U> = Pick<U, keyof T & keyof U>;
  */
 export type Overwrite<T, U> = Omit<T, keyof U> & U;
 
+/**
+ * a property P will be present if:
+ * - it is present in DecorationTargetProps
+ * - it is present in InjectedProps[P] and can satisfy DecorationTargetProps[P]
+ *   or it is not present in InjectedProps[P]
+ *
+ * source: @types/react-redux
+ */
+export type Matching<DecorationTargetProps, InjectedProps> = {
+  [P in keyof DecorationTargetProps]: P extends keyof InjectedProps
+    ? InjectedProps[P] extends DecorationTargetProps[P] ? DecorationTargetProps[P] : never
+    : DecorationTargetProps[P]
+};
```

Comment:
It seems like `Matching` is meant to accomplish much the same as `ConsistentWith`, but maybe does a better job? Do we still need `ConsistentWith` or can `Matching` replace it everywhere?

Reply:
I thought so too but `ConsistentWith` did not help with this issue. Maybe replace `ConsistentWith` in a future PR with `Matching` or explore where they actually differ. Same goes for `AnyComponent`.